### ISSUE TITLE
Error in stack frame diagram?

### ISSUE THREAD

**Issue by pixelshot91** [tnballo/high-assurance-rust#45]
In https://highassurance.rs/chp4/assure_stack_1.html#the-stack, the stack diagram show that the stack frame for the `square` function still exist under all the `recursive_count_down` stack frames.
https://github.com/tnballo/high-assurance-rust/blob/main/src/chp4/stack_example.svg

But in my understanding, the stack frame of `square` should pop before the stack frames of `recursive_count_down` are added.

Indeed if I print the backtrace with:
```rust
        println!("Boom!");
        println!("{:#?}", backtrace::Backtrace::capture());
        return x;
```
I get:

```
Backtrace [
    { fn: "stack_test::recursive_count_down", file: "./src/main.rs", line: 8 },
    { fn: "stack_test::recursive_count_down", file: "./src/main.rs", line: 13 },
    { fn: "stack_test::recursive_count_down", file: "./src/main.rs", line: 13 },
    { fn: "stack_test::recursive_count_down", file: "./src/main.rs", line: 13 },
    { fn: "stack_test::recursive_count_down", file: "./src/main.rs", line: 13 },
    { fn: "stack_test::main", file: "./src/main.rs", line: 35 },
...
```

Did I miss something ?

**Comment by tnballo** [comment]
Nope, you're not missing anything - this is indeed an error in the diagrams! 

`let _ = recursive_count_down(square(x));` does evaluate `square(x)` first and pop the frame, before calling `recursive_count_down`.

I'll fix the diagrams and add a blurb on printing the stack trace as you've described (seems there's no built-in environment variable to get a backtrace on stack overflow automatically, although the aptly-named [`backtrace-on-stack-overflow` crate ](https://crates.io/crates/backtrace-on-stack-overflow) can help debug this issue in larger programs.

**Comment by tnballo** [comment]
Also thank you for spotting this error :100:
